app/controllers/form_templates_controller.rb

```diff
@@ -23,4 +26,39 @@ def show
   def ensure_form_templates_enabled
     raise Discourse::InvalidAccess.new unless SiteSetting.experimental_form_templates
   end
+
+  def process_template(template_content)
+    parsed_template = YAML.safe_load(template_content)
+
+    parsed_template.map! do |form_field|
+      next form_field unless form_field["tag_group"]
+
+      tag_group_name = form_field["tag_group"]
+
+      tags =
+        TagGroup
+          .includes(:tags)
+          .visible(guardian)
+          .all
+          .where("lower(NAME) in (?)", tag_group_name.downcase)
+
+      ordered_field = {}
+
+      form_field.each do |key, value|
+        ordered_field[key] = value
+
+        ordered_field["choices"] = tags.first.tags.map(&:name) if key == "id"
+        if key == "attributes"
+          ordered_field["attributes"]["tag_group"] = tag_group_name
+          translated_tags =
+            tags.first.tags.select { |t| t.description }.to_h { |t| [t.name, t.description] }
+          ordered_field["attributes"]["tag_choices"] = translated_tags
+        end
+      end
+
+      ordered_field
+    end
+
+    YAML.dump(parsed_template)
+  end
```

Comment:
This is potentially a perf issue as we would have to do a query for each field here, we should improve this in a follow up query @jdmartinez1062

Reply:
I think its fixed now